cantools/database/can/conversion.py

```diff
@@ -0,0 +1,190 @@
+        self.is_float = is_float
+
+    def raw_to_scaled(
+        self,
+        raw_value: Union[int, float],
+        decode_choices: bool = True,
+    ) -> Union[int, float]:
+        return raw_value
+
+    def scaled_to_raw(self, scaled_value: SignalValueType) -> Union[int, float]:
+        if not isinstance(scaled_value, (int, float)):
+            raise TypeError(
+                f"'scaled_value' must have type 'int' or 'float' (is {type(scaled_value)})"
+            )
+        return scaled_value
+
+
+class LinearIntegerConversion(BaseConversion):
+    is_float = False
+    choices = None
+
+    def __init__(self, scale: int, offset: int) -> None:
+        self.scale: int = scale
+        self.offset: int = offset
+
+    def raw_to_scaled(
+        self,
+        raw_value: Union[int, float],
+        decode_choices: bool = True,
+    ) -> SignalValueType:
+        return raw_value * self.scale + self.offset
+
+    def scaled_to_raw(self, scaled_value: SignalValueType) -> Union[int, float]:
+        if not isinstance(scaled_value, (int, float)):
+            raise TypeError(
+                f"'scaled_value' must have type 'int' or 'float' (is {type(scaled_value)})"
+            )
+        return round((scaled_value - self.offset) / self.scale)
+
+
+class LinearConversion(BaseConversion):
+    choices = None
+
+    def __init__(self, scale: float, offset: float, is_float: bool) -> None:
+        self.scale = scale
+        self.offset = offset
+        self.is_float = is_float
+
+    def raw_to_scaled(
+        self,
+        raw_value: Union[int, float],
+        decode_choices: bool = True,
+    ) -> SignalValueType:
+        return raw_value * self.scale + self.offset
+
+    def scaled_to_raw(self, scaled_value: SignalValueType) -> Union[int, float]:
+        if not isinstance(scaled_value, (int, float)):
+            raise TypeError(
+                f"'scaled_value' must have type 'int' or 'float' (is {type(scaled_value)})"
+            )
+        _raw = (scaled_value - self.offset) / self.scale
+        if self.is_float:
+            return _raw
+        return round(_raw)
+
+
+class NamedSignalConversion(BaseConversion):
+    is_float = False
+
+    def __init__(self, scale: float, offset: float, choices: Choices) -> None:
+        self.scale = scale
+        self.offset = offset
+        self._inverse_choices: Optional[Dict[str, int]] = None
+        self.choices = choices
+        self._update_choices()
+
+        self._conversion = conversion_factory(
+            scale=self.scale,
+            offset=self.offset,
+            choices=None,
+            is_float=False,
+        )
+
+    def raw_to_scaled(
+        self,
+        raw_value: Union[int, float],
+        decode_choices: bool = True,
+    ) -> SignalValueType:
+        if decode_choices:
+            try:
+                return self.choices[raw_value]  # type: ignore[index]
+            except (AttributeError, KeyError):
+                pass
```

Comment:
to make debugging easier, I prefer not to have exceptions in the happy path.
```suggestion
        if (choice := self.choices.get(raw_value)) is not None:
            return choice
```
(possibly, this even might run slightly faster...)

Reply:
I was curious about this, so i measured it 😄 

```python3
import timeit


class MyClass:
    choices = {
        1: "1",
    }

    def raw_to_scaled1(self, raw_value):
        try:
            return self.choices[raw_value]  # type: ignore[index]
        except (AttributeError, KeyError):
            pass

        return None

    def raw_to_scaled2(self, raw_value):
        if (choice := self.choices.get(raw_value)) is not None:
            return choice

        return None


instance = MyClass()

n = 100_000
dt = timeit.repeat('instance.raw_to_scaled1(1)', globals=globals(), repeat=20, number=n)
print(f"{min(dt)/n * 1e6} us - try-except happy path")
dt = timeit.repeat('instance.raw_to_scaled1(2)', globals=globals(), repeat=20, number=n)
print(f"{min(dt)/n * 1e6} us - try-except happy miss")

dt = timeit.repeat('instance.raw_to_scaled2(1)', globals=globals(), repeat=20, number=n)
print(f"{min(dt)/n * 1e6} us - walrus happy path")
dt = timeit.repeat('instance.raw_to_scaled2(2)', globals=globals(), repeat=20, number=n)
print(f"{min(dt)/n * 1e6} us - walrus happy miss")
```

Python 3.11:
```
0.05078900000171416 us - try-except happy path
0.2410190000000512 us - try-except happy miss
0.0671630000033474 us - walrus happy path
0.06490199999916513 us - walrus happy miss
```

so yeah, yours looks better 👍